src/var/getShadowRoot.js

```diff
@@ -0,0 +1,22 @@
+define( [
+	"./document"
+], function( document ) {
+	"use strict";
+
+	return function getShadowRoot( elem ) {
+
+		// Check if browser supports Shadow DOM
+		if ( document.head.createShadowRoot || document.head.attachShadow ) {
```

Comment:
Why do we need to check both?

Reply:
This `if` is still in the function so it still executes every time `isAttached` is invoked. I think @gibson042 meant to define this function conditionally, to run the Shadow DOM support test outside of the function and assign a proper definition based on that. In this way this condition will fire only once.

Reply:
So to get it clear, you mean this should be written as a condition wherever isAttached is invoked and send some parameter based on that?

Reply:
> Why do we need to check both?

So `createShawowRoot` gives a false negative in safari version 10.1 since it uses 0.v1 of shadowDOM element which had `attachShadow`. So I included both.

Reply:
I don't know what's @gibson042's opinion on that but personally I wouldn't botter with supporting Shadow DOM v0; it was a preview and the standard is the v1 version. And if we add v0 support we won't be able to remove it so easily later.

Reply:
> So to get it clear, you mean this should be written as a condition wherever isAttached is invoked and send some parameter based on that?

Something like:
```js
var myFunction;
if (someCondition) {
    myFunction = () => someValue;
} else {
    myFunction = () => someOtherValue;
```
(but in ES5 etc., this is just to get an idea) Then the condition is only checked once and is never checked when `myFunction` is getting invoked.


We did something in this spirit for some `cssHooks`, see:
https://github.com/jquery/jquery/blob/3.3.1/src/css/addGetHookIf.js
and its uses:
https://github.com/jquery/jquery/blob/3.3.1/src/css.js#L414-L425
https://github.com/jquery/jquery/blob/3.3.1/src/offset.js#L218-L229

Reply:
@mgol isn't in the above examples, in https://github.com/jquery/jquery/blob/3.3.1/src/css/addGetHookIf.js#L10 the condition is checked everytime the `addGetHookIf` is invoked?

Reply:
`addGetHookIf` is only called once per each css hook used.